src/schoology-extractor/schoology_extractor/main.py

```diff
@@ -3,76 +3,128 @@
+
+
+logger.info("Starting Ed-Fi LMS Schoology Extractor")
+logger.debug("Loading and processing .env variables")
+
 schoology_key = os.getenv("SCHOOLOGY_KEY")
 schoology_secret = os.getenv("SCHOOLOGY_SECRET")
 schoology_output_path = os.getenv("SCHOOLOGY_OUTPUT_PATH")
 schoology_grading_periods = os.getenv("SCHOOLOGY_GRADING_PERIODS")
 
-assert schoology_key is not None, "A `SCHOOLOGY_KEY` must be present in the .env file and it was not found."
-assert schoology_secret is not None, "A `SCHOOLOGY_SECRET` must be present in the .env file and it was not found."
-assert schoology_output_path is not None, "A `SCHOOLOGY_OUTPUT_PATH` must be present in the .env file and it was not found."
-assert schoology_grading_periods is not None, "A `SCHOOLOGY_GRADING_PERIODS` must be present in the .env file and it was not found."
+assert (
+    schoology_key is not None
+), "A `SCHOOLOGY_KEY` must be present in the .env file and it was not found."
+assert (
+    schoology_secret is not None
+), "A `SCHOOLOGY_SECRET` must be present in the .env file and it was not found."
+assert (
+    schoology_output_path is not None
+), "A `SCHOOLOGY_OUTPUT_PATH` must be present in the .env file and it was not found."
+assert (
+    schoology_grading_periods is not None
+), "A `SCHOOLOGY_GRADING_PERIODS` must be present in the .env file and it was not found."
+
+
+grading_periods_array = schoology_grading_periods.split(",")
 
+logger.debug("Finished loading and processing .env variables")
 
-grading_periods_array = schoology_grading_periods.split(',')
 
 request_client = RequestClient(schoology_key, schoology_secret)
 
 # export users
+logger.info("Exporting users")
 users_response = request_client.get_users()
-users_list = []
+users_list: List[Any] = []
 while True:
     users_list = users_list + users_response.current_page_items
     if users_response.get_next_page() is None:
         break
 
-export_data.to_csv(users_list, os.path.join(schoology_output_path, 'users.csv'))
+export_data.to_csv(users_list, os.path.join(schoology_output_path, "users.csv"))
+logger.info("Finished exporting users")
 
 # export sections
+logger.info("Exporting sections")
 
 # first we need to get a list of courses
+logger.info("Exporting sections - Getting courses")
 courses_response = request_client.get_courses()
-courses_list = []
+courses_list: List[Any] = []
 while True:
     courses_list = courses_list + courses_response.current_page_items
     if courses_response.get_next_page() is None:
         break
+logger.info("Exporting sections - Finished getting courses")
 
 # now we can get a list of sections
 course_ids = map(lambda x: x["id"], courses_list)
 sections_list = request_client.get_section_by_course_ids(list(course_ids))
-export_data.to_csv(sections_list,
-                   os.path.join(schoology_output_path, 'sections.csv'))
+export_data.to_csv(sections_list, os.path.join(schoology_output_path, "sections.csv"))
+logger.info("Finished exporting sections")
```

Comment:
Is the presence of the next info statement good enough? Perhaps we don't need line 91.

Reply:
Removed ✅